CondCore/ESSources/test/TestConcurrentIOVsCondCore_cfg.py

```diff
@@ -0,0 +1,78 @@
+# This configuration tests CondDBESSource and the EventSetup
+# system. It causes it to read BeamSpot data objects from
+# the database. The global tag and run number are selected
+# to access this data in a run which is known to have
+# multiple IOVs in the same run. This is a very simple job
+# with empty input and one event per luminosity block.
+# The analyzer gets this BeamSpot object and prints
+# values from it to standard out. This is compared to
+# a reference file of known expected files. If the
+# correct data is not read and written to output then
+# the unit test fails. The additional wrinkle here is that
+# the analyzer puts in a "busy wait" (intentional does
+# some calculation that uses CPU for a while). It does
+# this for each event that is the first event of a
+# new IOV.  That forces multiple IOVs to be running
+# concurrently. Further the delay decreases as we move
+# forward so the data gets are not occurring in the same
+# order as the lumis are being input. The purpose of
+# all this is to test that CondDBESSource and the
+# EventSetup system can process IOVs concurrently
+# and deliver the same data objects as when processing
+# one IOV at a time.
+
+import FWCore.ParameterSet.Config as cms
+
+process = cms.Process("TEST")
+
+#process.Tracer = cms.Service("Tracer")
+
+process.options = dict(
+    numberOfThreads = 4,
+    numberOfStreams = 4,
+    numberOfConcurrentRuns = 1,
+    numberOfConcurrentLuminosityBlocks = 4,
+    eventSetup = dict(
+        numberOfConcurrentIOVs = 4
+    )
+)
+
+process.load("FWCore.MessageService.MessageLogger_cfi")
+#process.MessageLogger.cerr.threshold = 'INFO'
+#process.MessageLogger.cerr.INFO.limit = 1000000
+
+process.source = cms.Source("EmptySource",
+    firstRun = cms.untracked.uint32(132598),
+    firstLuminosityBlock = cms.untracked.uint32(1),
+    firstEvent = cms.untracked.uint32(1),
+    numberEventsInRun = cms.untracked.uint32(1000000),
+    numberEventsInLuminosityBlock = cms.untracked.uint32(1)
+)
+
+process.maxEvents = cms.untracked.PSet(
+  input = cms.untracked.int32(200)
+)
+
+process.GlobalTag = cms.ESSource("PoolDBESSource",
+    DBParameters = cms.PSet(
+        authenticationPath = cms.untracked.string(''),
+        authenticationSystem = cms.untracked.int32(0),
+        messageLevel = cms.untracked.int32(0),
+        security = cms.untracked.string('')
+    ),
+    DumpStat = cms.untracked.bool(False),
+    ReconnectEachRun = cms.untracked.bool(False),
+    RefreshAlways = cms.untracked.bool(False),
+    RefreshEachRun = cms.untracked.bool(False),
+    RefreshOpenIOVs = cms.untracked.bool(False),
+    connect = cms.string('frontier://FrontierProd/CMS_CONDITIONS'),
+    globaltag = cms.string('110X_dataRun2_v5'),
```

Comment:
@wddgit @christopheralanwest @ggovi 
I am wondering if it is intentional to hardcode a GT name here. 
Whenever a record present in this GT will be removed from CMSSW, that will cause errors in this unit test (see e.g.: https://github.com/cms-sw/cmssw/pull/31729#issuecomment-708664358).
For this use case, it seems to me that's an overkill. Would not it be sufficient to just access on multi-IOVed per-LS record BeamSpot tag such as in:
```diff
diff --git a/CondCore/ESSources/test/TestConcurrentIOVsCondCore_cfg.py b/CondCore/ESSources/test/TestConcurrentIOVsCondCore_cfg.py
index c6aed33355c..7cfecb62e9b 100644
--- a/CondCore/ESSources/test/TestConcurrentIOVsCondCore_cfg.py
+++ b/CondCore/ESSources/test/TestConcurrentIOVsCondCore_cfg.py
@@ -66,12 +66,15 @@ process.GlobalTag = cms.ESSource("PoolDBESSource",
     RefreshEachRun = cms.untracked.bool(False),
     RefreshOpenIOVs = cms.untracked.bool(False),
     connect = cms.string('frontier://FrontierProd/CMS_CONDITIONS'),
-    globaltag = cms.string('110X_dataRun2_v5'),
+    globaltag = cms.string(''),
     pfnPostfix = cms.untracked.string(''),
     pfnPrefix = cms.untracked.string(''),
-    snapshotTime = cms.string(''),
-    toGet = cms.VPSet()
-)
+    #snapshotTime = cms.string(''),
+    snapshotTime = cms.string('2020-10-10 00:00:00.000'),
+    toGet = cms.VPSet(cms.VPSet(cms.PSet(record = cms.string("BeamSpotObjectsRcd"),
+                                         tag = cms.string("BeamSpotObjects_2017UL_LumiBased_v2")
+                                         )))
+    )
```
?
IFF there are concerns that the tag content might change over time (unlikely, but possible due to accidental overwritings) I believe one can leverage the `snapshotTime` of the `ESSource`.

Reply:
I'll submit a pull request to fix this. Thanks for debugging this. I did not know global tags could become obsolete in this way. As long as your changes lead to the same data being read from the database for run 132598 (selected because I know the values in the database are different for different lumis), then it should work just fine for this unit test. @mmusich Do you prefer the empty snapshotTime string or the non-empty one above?

Reply:
Thanks.
> Do you prefer the empty snapshotTime string or the non-empty one above?

Probably non-empty is safer, but the ultimate decision should come from @ggovi